pkg/alertmanager/clustertlsconfig/config.go

```diff
@@ -0,0 +1,300 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clustertlsconfig
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	webconfig "github.com/prometheus-operator/prometheus-operator/pkg/webconfig"
+)
+
+const (
+	cmdflag            = "cluster.tls-config"
+	volumeName         = "cluster-tls-config"
+	serverVolumePrefix = "cluster-tls-server-config-"
+	clientVolumePrefix = "cluster-tls-client-config-"
+	serverTLSCredDir   = "server_tls"
+	clientTLSCredDir   = "client_tls"
+	ConfigFileKey      = "cluster-tls-config.yaml"
+)
+
+// Config is the web configuration for prometheus and alertmanager instance.
+//
+// Config can make a secret which holds the web config contents, as well as
+// volumes and volume mounts for referencing the secret and the
+// necessary TLS credentials.
+type Config struct {
+	clusterTLSConfig    *monitoringv1.ClusterTLSConfig
+	serverTLSReferences *webconfig.TLSReferences
+	clientTLSReferences *webconfig.TLSReferences
+	mountingDir         string
+	secretName          string
+}
+
+// New creates a new ClusterTLSConfig.
+// All volumes related to the cluster TLS config will be mounted via the `mountingDir`.
+// The Secret where the cluster TLS config will be stored will be named `secretName`.
+// All volumes containing TLS credentials related to cluster TLS configuration will be prefixed with "cluster-tls-server-config-"
+// or "cluster-tls-client-config-" respectively, for server and client credentials.
+func New(mountingDir string, secretName string, clusterTLSConfig *monitoringv1.ClusterTLSConfig) (*Config, error) {
+	if clusterTLSConfig == nil {
+		return &Config{
+			mountingDir: mountingDir,
+			secretName:  secretName,
+		}, nil
+	}
+
+	var (
+		clientTLSCreds *webconfig.TLSReferences
+		serverTLSCreds *webconfig.TLSReferences
+	)
+
+	serverTLSConfig := clusterTLSConfig.ServerTLS
+	if err := serverTLSConfig.Validate(); err != nil {
+		return nil, err
+	}
+
+	clientTLSConfig := clusterTLSConfig.ClientTLS
+	if err := clientTLSConfig.Validate(); err != nil {
+		return nil, err
+	}
+
+	serverTLSCreds = webconfig.NewTLSReferences(path.Join(mountingDir, serverTLSCredDir), serverTLSConfig.KeySecret, serverTLSConfig.Cert, serverTLSConfig.ClientCA)
+	clientTLSCreds = webconfig.NewTLSReferences(path.Join(mountingDir, clientTLSCredDir), *clientTLSConfig.KeySecret, clientTLSConfig.Cert, clientTLSConfig.CA)
+
+	return &Config{
+		clusterTLSConfig:    clusterTLSConfig,
+		serverTLSReferences: serverTLSCreds,
+		clientTLSReferences: clientTLSCreds,
+		mountingDir:         mountingDir,
+		secretName:          secretName,
```

Comment:
(nit) can we remove the secretName field? It doesn't seem to be used anywhere in this package.

Reply:
We do make use of it in the `makeVolume()` function and to while creating the secret in `createOrUpdateClusterTLSConfigSecret()`. We could make it a constant. I think we had a discussion about this offline and chose to stick with this (mainly because it wasn't a big difference either way). 
WDYT?